engine/consensus/approvals/assignment_collector_tree.go

```diff
@@ -266,6 +266,11 @@ func (t *AssignmentCollectorTree) GetOrCreateCollector(result *flow.ExecutionRes
 	if err != nil {
 		return nil, fmt.Errorf("failed to store assignment collector into the tree: %w", err)
 	}
+	// leveled forest doesn't treat this case as error, we shouldn't create collectors
+	// for vertexes lower that forest.LowestLevel
+	if vertex.Level() < t.forest.LowestLevel {
+		return nil, fmt.Errorf("could not add collector with height lower than the lowest level")
+	}
```

Comment:
Agree with Leo:
to me it would also be more intuitive if this check happened right after  https://github.com/onflow/flow-go/blob/321088900138ad05e010cd78333f05bad7dd5368/engine/consensus/approvals/assignment_collector_tree.go#L254-L255